core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java

```diff
@@ -355,26 +355,46 @@ void removeBuddyInfoForRemovedUser(User user, Buddy buddy)
 		if (buddy.getSendingStatus() == Status.ACCEPTED || buddy.getReceivingStatus() == Status.ACCEPTED)
 		{
 			// Buddy request was accepted
-			removeMessagesSentByBuddy(user, buddy);
-			removeMessagesSentByUserToBuddy(user, buddy);
-
-			// Send message to "self", to notify the user about the removed buddy user
-			UUID buddyUserAnonymizedId = getUserAnonymizedIdForBuddy(buddy);
-			sendDropBuddyMessage(BuddyInfoParameters.createInstance(buddy, buddyUserAnonymizedId), Optional.empty(),
-					DropBuddyReason.USER_ACCOUNT_DELETED, user);
+			prepareCleanupRemovedBuddyInAcceptedState(user, buddy);
 		}
 		else if (buddy.getSendingStatus() != Status.REJECTED && buddy.getReceivingStatus() != Status.REJECTED)
 		{
 			// Buddy request was not accepted or rejected yet
-			// Send message to "self", as if the requested user declined the buddy request
-			UUID buddyUserAnonymizedId = buddy.getUserAnonymizedId().orElse(null); //
-			sendBuddyConnectResponseMessage(BuddyInfoParameters.createInstance(buddy, buddyUserAnonymizedId),
-					user.getUserAnonymizedId(), buddy.getId(), user.getDevices(), Status.REJECTED,
-					getDropBuddyMessage(DropBuddyReason.USER_ACCOUNT_DELETED, Optional.empty()));
+			prepareCleanupRemovedBuddyInPendingState(user, buddy);
 		}
 		removeBuddy(user.getId(), buddy);
 	}
 
+	private void prepareCleanupRemovedBuddyInPendingState(User user, Buddy buddy)
+	{
+		messageService.deleteUnprocessedMessages(user, m -> isBuddyConnectResponseFromBuddy(m, buddy));
```

Comment:
It *might* occur that at the same time, the buddy status is changed to accepted, isn't it? So shouldn't we to be safe remove all messages from the buddy, not only the unprocessed ones?

Reply:
No, this cannot occur. The current user would need to switch it to accepted, but that is impossible for a buddy that overwrote their account.